### PR TITLE
rename `kubernetes-master` applications to `kubernetes-control-plane`

### PR DESCRIPTION
**Pull request by addyess**
In each bundle, renamed `kubernetes-master` application to `kubernetes-control-plane`
In each fragment, update application references
In each overlay, update application references
Each README has changed the word master to control-plane
* merged with https://github.com/charmed-kubernetes/jenkins/pull/802

### CODE REVIEW

**Review by stonepreston — APPROVED** [review]
LGTM

**Comment by addyess** [comment]
Let's hold this one for next week.  No rush here

File: bundle [hunk 1]
```diff
@@ -36,6 +36,8 @@ class CharmstoreQueryError(Exception):
 def get_args():
     ''' Parse cli arguments. '''
     parser = argparse.ArgumentParser()
+    parser.add_argument('-n', '--name', default=None,
+                        help='name given to the created bundle.')
```
Comment (addyess) [inline bundle:40]: a valid `bundle.yaml` must have a name.  If you don't provide one, one will be generated based on the path name

File: bundle [hunk 2]
```diff
@@ -128,29 +140,33 @@ def version_bundle(bundle, channel, overrides):
...
+                resp = requests.get('https://api.jujucharms.com/charmstore/v5/meta/id',
+                                    params={'id': charm, 'channel': channel})
+                if not resp.ok:
+                    err = "Couldn't query %s %s, is it public?" % (charm, channel)
+                    raise CharmstoreQueryError(err)
+                meta = resp.json()
+                meta = [meta[k] for k in meta][0]
+                copy['applications'][application]['charm'] = meta['Id']
+            else:
+                copy['applications'][application]['channel'] = channel
```
Comment (addyess) [inline bundle:153]: each ~containers owned charm in charmhub (ie. `1.24/edge`) will be up-to-date based on its channel.  Therefore there's no need to use specific revisions

File: bundle [hunk 3]
```diff
@@ -128,29 +140,33 @@ def version_bundle(bundle, channel, overrides):
...
+                                params={'id': charm, 'channel': channel})
+            if not resp.ok:
+                err = "Couldn't query resources for %s %s, is it public?" % (charm, channel)
+                raise CharmstoreQueryError(err)
+            response_data = resp.json()
+            copy['applications'][application].setdefault('resources', {})
+            for resource in response_data[charm]:
+                name = resource['Name']
+                revision = resource['Revision']
+                copy['applications'][application]['resources'][name] = revision
```
Comment (addyess) [inline bundle:169]: charmhub charms have revisions pinned per track -- therefore it's not necessary to specify those revisions in the bundle

**Review by stonepreston — APPROVED** [review]
LGTM

**Pull request merged**